llvm/lib/MC/MCCASObjectV1.cpp

```diff
@@ -936,6 +937,34 @@ Error MCCASBuilder::createStringSection(
   return Error::success();
 }
 
+Error MCCASBuilder::createLineSection(
+    StringRef S, std::function<Error(StringRef)> CreateFn) {
+  BinaryStreamReader SectionReader(S, Asm.getBackend().Endian);
+  while (true) {
+    uint32_t Length;
+    uint64_t SectionStartOffset = SectionReader.getOffset();
+    if (auto E = SectionReader.readInteger(Length))
+      return E;
+    if (Length != 0xffffffff) {
+      // The Debug Information is 32-bit DWARF
+      if (auto E = SectionReader.skip(Length))
+        return E;
+    } else {
+      uint64_t ExtendedLength;
+      if (auto E = SectionReader.readInteger(ExtendedLength))
+        return E;
+      if (auto E = SectionReader.skip(ExtendedLength))
+        return E;
+    }
```

Comment:
Should we make this a helper function? I'll need it too. This is what I have right now:

```cpp
/// Reads the Length field of the debug info header contained in Reader,
/// assuming Reader is positioned at the beginning of the section.
/// The reader state is advanced by however many bytes are used by the Length
/// field, plus Length bytes.
Expected<size_t> getSizeFromDebugInfoHeaderAndSkip(BinaryStreamReader &Reader) {
  // From Dwarf 5 section 7.4:
  // In the 32-bit DWARF format, an initial length field [...] is an unsigned
  // 4-byte integer (which must be less than 0xfffffff0);
  uint32_t Word1;
  if (auto E = Reader.readInteger(Word1))
    return E;

  // TODO: handle 64-bit DWARF format.
  assert(Word1 < 0xfffffff0);
  return Word1;
}
```


And then later, I use this function like this:

```cpp
  while (!Reader.empty()) {
    auto StartOffset = Reader.getOffset();
    auto Length = getSizeFromDebugInfoHeaderAndSkip(Reader);
    if (!Length)
      return Length.takeError();

    // TODO: zero out abbrev offset once it's been split
    auto SectionData =
        DebugInfoStrRef.slice(StartOffset, Reader.getOffset() - StartOffset);
    auto DbgInfoRef = DebugInfoRef::create(*this, SectionData);

```

Reply:
Yes that can be done, I will change it